packages/enzyme/src/ShallowWrapper.js

```diff
@@ -369,7 +369,39 @@ class ShallowWrapper {
     }
     this.single('setState', () => {
       withSetStateAllowed(() => {
-        this.instance().setState(state, callback);
+        const adapter = getAdapter(this[OPTIONS]);
+        const instance = this.instance();
+        const prevProps = instance.props;
+        const prevState = instance.state;
+        const prevContext = instance.context;
+        let shouldRender = true;
+        // dirty hack: avoid calling shouldComponentUpdate twice
```

Comment:
why is this needed? can you elaborate in the comment?

Reply:
Added a comment. Does this make sense?